src/widgets/openmediavault/OpenMediaVaultTile.tsx

```diff
@@ -0,0 +1,119 @@
+import { Box, Flex, Group, Paper, ScrollArea, Text } from '@mantine/core';
+import { IconInfoSquare } from '@tabler/icons-react';
+import { useTranslation } from 'next-i18next';
+import { useConfigContext } from '~/config/provider';
+import { api } from '~/utils/api';
+
+import { defineWidget } from '../helper';
+import { WidgetLoading } from '../loading';
+import { IWidget } from '../widgets';
+import OpenMediaVaultCpu from './OpenMediaVaultCpu';
+import OpenMediaVaultFileSystem from './OpenMediaVaultFileSystem';
+import OpenMediaVaultMemory from './OpenMediaVaultMemory';
+
+const definition = defineWidget({
+  id: 'openmediavault',
+  icon: 'https://cdn.jsdelivr.net/gh/walkxcode/dashboard-icons/png/openmediavault.png',
+  options: {},
+  gridstack: {
+    minWidth: 1,
+    minHeight: 1,
+    maxWidth: 8,
+    maxHeight: 5,
+  },
+  component: OpenMediaVaultWidgetTile,
+});
+
+export type IOpenMediaVaultWidget = IWidget<(typeof definition)['id'], typeof definition>;
+
+interface OpenMediaVaultWidgetProps {
+  widget: IOpenMediaVaultWidget;
+}
+function OpenMediaVaultWidgetTile({ widget }: OpenMediaVaultWidgetProps) {
+  const { t } = useTranslation('modules/openmediavault');
+  const { isInitialLoading, data: auth } = useOpenmediavaultAuthQuery();
+  const { data: info } = useOpenmediavaultInfoQuery();
+  const { data: fileSystem } = useOpenmediavaultFileSystemQuery();
+  const { data: cpuTemp } = useOpenmediavaultCpuTempQuery();
+
+  if (isInitialLoading || !auth || !info || !fileSystem || !cpuTemp) {
+    return <WidgetLoading />;
+  }
+
+  const formatUptime = (uptime: number) => {
+    const days = Math.floor(uptime / 60 / 60 / 24);
+    const remainingHours = Math.floor((uptime % (days * 24 * 60 * 60)) / 60 / 60);
+    return `${days} days & ${remainingHours} hours`;
+  };
```

Comment:
Extract to a function in the helpers?

Reply:
What do you mean?

Reply:
I would move this function outside the widgets to the tools. We have files with these kind of functions